### DISCUSSION TITLE
MarkupExtension to allow two way binding a BehaviorSubject/Subject

### DISCUSSION THREAD

**Discussion by JaggerJo** [AvaloniaUI/Avalonia#14174] [Q&A]
As the title explains I want to create a markup extension to go full circle with observable support. Being able to bind to `IObservable<T>s` is already a great feature. However for properties like `SelectedValue`, `Text`, .. it would be great if bindings would work in both directions. 

I'd like to implement this as a Compiled Binding, but don't know how to approach this. Do I start by inheriting `CompiledBindingExtension`?

**Comment by JaggerJo** [comment]
Going from a `Subject<T>` / `BehaviorSubject<T>` to an InstanceBinding is easy. But how do I event get the passed Subject?

```csharp
var _subject = new Subject<object?>()
InstancedBinding.TwoWay(_subject, _subject)
```

**Comment by timunie** [comment]
worth to check out binding refractor PR here

**Reply by JaggerJo** [reply]
Thanks for the hint.

Link for reference: https://github.com/AvaloniaUI/Avalonia/pull/13970
